bson/src/main/org/bson/json/JsonScanner.java

```diff
@@ -115,9 +119,8 @@ public JsonToken nextToken() {
      */
     private JsonToken scanRegularExpression() {
 
-        int start = buffer.getPosition() - 1;
-        int options = -1;
-
+        StringBuilder patternBuilder = new StringBuilder();
```

Comment:
Changes to scan* methods are required because they currently treat the buffer as something that can be accessed randomly.  These changes make it so they work with a streaming buffer.

Reply:
ACK